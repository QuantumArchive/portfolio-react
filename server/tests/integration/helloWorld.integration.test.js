const expect = require('expect')
const supertest = require('supertest')
const { server } = require('../../index')

describe('/server/resolvers/helloWorld', () => {
  const request = supertest(server)
  const graphQlEndpoint = '/graphql'
  const timeout = 15000

  afterAll((done) => {
    server.close(done)
  }, timeout)

  test('returns a body with hello world on it', async () => {
    const { body: { data: { greeting } } } = await request.post(graphQlEndpoint).send({
      query: `query Greeting {
        greeting {
          message
        }
      }`,
      variables: {}
    })
    expect(greeting).toMatchSnapshot()
  })
})
