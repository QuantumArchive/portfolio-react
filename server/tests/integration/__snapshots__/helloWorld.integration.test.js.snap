// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/server/resolvers/helloWorld returns a body with hello world on it 1`] = `
Object {
  "message": "hello world",
}
`;
